subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/NormalizingJavaCompiler.java

```diff
@@ -88,7 +89,7 @@ private void logCompilerArguments(JavaCompileSpec spec) {
         }
 
         List<String> compilerArgs = new JavaCompilerArgumentsBuilder(spec).includeLauncherOptions(true).includeSourceFiles(true).build();
-        String joinedArgs = Joiner.on(' ').join(compilerArgs);
+        String joinedArgs = compilerArgs.stream().map(it -> it != null && StringUtils.isBlank(it) ? ('"' + it + '"') : it).collect(Collectors.joining(" "));
```

Comment:
I think the null check here is unnecessary, I don't think nulls can be present in the built list.  They would have shown up as "null" in the output previously if they were, which I think we would have fixed.

Reply:
Interestingly though, some [existing code](https://github.com/gradle/gradle/blob/eaafa52cf420922c8d0a8b2eb7e4c4a40812a6eb/subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/JavaCompilerArgumentsBuilder.java#L109) also guards against nulls in the compiler args list. But you are right that that is pointless, as nulls would trigger NPEs when we [convert arg Objects to Strings](https://github.com/gradle/gradle/blob/eaafa52cf420922c8d0a8b2eb7e4c4a40812a6eb/subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/JavaCompilerArgumentsBuilder.java#L83) earlier in the code.

@tresat  I can happily take that check out. If so, does that affect the approval state of this PR, or could I still request bot-gradle to test and merge it?

Reply:
Added a commit that clarifies the contract for `JavaCompilerArgumentsBuilder` and removed the unnecessary null check

Reply:
@abstratt Additional change shouldn't affect the approval state.  I've been requesting re-reviews for major changes, but wouldn't for something like this.